sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -2735,6 +2735,29 @@ abstract class CSVSuite
       }
     }
   }
+
+  test("SPARK-38520: Intervals are overflow") {
```

Comment:
The issue is not specific to the CSV datasource. Could you move/port tests to `IntervalUtilsSuite`.

Reply:
moved to IntervalUtilsSuite